### PR TITLE
Removed skipped unit test and the use of eval.  Enhancement

### PR DESCRIPTION
**Pull request by levelfivehub**
Hi,

This is my second contribution towards ZF2.  I hope it suits this module well.

There was a skipped test (commented out) as well as the use of eval().  I did notice a comment with a warning so thought I put my contribution forward.  Please review and let me know if it is acceptable.

Awaiting your comments.  Unit tests and builds passing.

Also did php-cs-fixes.

Ps.  There may be other scenarios I can cover in this branch.  But ideally I want to get some feedback before continuing.

### CODE REVIEW

**Comment by Maks3w** [comment]
Please use different branches on your fork and don't make everything in the same PR.

http://akrabat.com/the-beginners-guide-to-contributing-to-a-github-project/

File: src/Adapter/PhpCode.php [hunk 1]
```diff
@@ -9,52 +9,121 @@
...
-            }
+    /**
+     * Check if string is already serialized
+     *
+     * @param mixed $code
+     * @return bool
+     */
+    private function isSerialized($code)
+    {
+        $unserialize = @unserialize($code);
```
Comment (Maks3w) [inline src/Adapter/PhpCode.php:121]: setup an error handler instead

File: src/Adapter/Json.php [hunk 2]
```diff
@@ -45,6 +45,7 @@ public function getOptions()
         if ($this->options === null) {
             $this->options = new JsonOptions();
         }
+
```
Comment (Maks3w) [inline src/Adapter/Json.php:48]: This kind of cosmetic changes should be avoided

**Comment by levelfivehub** [comment]
@Maks3w  In relation to your comment about branches, what you recommend for this one?

File: src/Adapter/Json.php (hunk 2 above)
Comment (levelfivehub) [inline src/Adapter/Json.php:48]: I can revert.  I thought as I was looking at the file, I add the space for consistency.

File: test/Adapter/PhpCodeTest.php [hunk 3]
```diff
@@ -31,106 +31,117 @@ public function tearDown()
...
     {
-        if (version_compare(PHP_VERSION, '7', 'ge')) {
-            $this->markTestSkipped('Cannot catch parse errors in PHP 7+');
-        }
-        $value = 'not a serialized string';
-
-        $this->setExpectedException('Zend\Serializer\Exception\RuntimeException', 'syntax error');
-        $this->adapter->unserialize($value);
+        return [
+            [
```
Comment (Maks3w) [inline test/Adapter/PhpCodeTest.php:72]: Use array keys for give a name to each data set

File: test/Adapter/PhpCodeTest.php [hunk 4]
```diff
@@ -31,106 +31,117 @@ public function tearDown()
...
+                'expected' => serialize('test'),
+                'serialize' => true
+            ],
+            [
+                'value' => '<?php echo "test"; ?>',
+                'expected' => serialize('<?php echo "test"; ?>'),
+                'serialize' => true
+            ],
+            [
+                'value' => 'test',
```
Comment (Maks3w) [inline test/Adapter/PhpCodeTest.php:83]: Duplicated

Comment (levelfivehub) [inline test/Adapter/PhpCodeTest.php:83]: Good catch, thanks.  Will remove.

File: test/Adapter/PhpCodeTest.php (hunk 3 above)
Comment (Maks3w) [inline test/Adapter/PhpCodeTest.php:72]: I think this test is more legible if formated as columns

Ex: https://github.com/zendframework/zend-inputfilter/blob/a05fa0bfca3a82ef119e1af2e970a48d787913c7/test/InputTest.php#L572-L589

**Comment by Maks3w** [comment]
Put the coding style and dockblocks changes in a different PR. Those PRs will be merged quickly are less risky.

File: test/Adapter/PhpCodeTest.php (hunk 3 above)
Comment (levelfivehub) [inline test/Adapter/PhpCodeTest.php:72]: I like that example.  I will put it into practice.

File: src/Adapter/PhpCode.php (hunk 1 above)
Comment (levelfivehub) [inline src/Adapter/PhpCode.php:121]: I will give it a go :+1:

**Comment by levelfivehub** [comment]
@Maks3w 
I will put those styling and doc block changes in a separate PR.

**Comment by Maks3w** [comment]
1. While this PR is good for master, exceptionally use develop branch because that branch will be released soon and we can avoid merge conflicts.
2. If you will try to increase the test coverage please do a PR like this first https://github.com/zendframework/zend-inputfilter/commit/60434bc2ed16efacbdef57f269cf5b1d8817bfc2

**Comment by levelfivehub** [comment]
@Maks3w 

Would you suggest branching off develop and cherry-pick?

**Comment by Maks3w** [comment]
Yes.

**Comment by levelfivehub** [comment]
@Maks3w  Stand by.  I will attempt to get this ready for today.

**Pull request closed without merging**